state-chain/pallets/cf-ingress-egress/src/tests.rs

```diff
@@ -1474,3 +1461,35 @@ fn consolidation_tx_gets_broadcasted_on_finalize() {
 		));
 	});
 }
+
+#[test]
+fn handle_prewitness_deposit() {
+	const ASSET: cf_chains::assets::eth::Asset = eth::Asset::Eth;
+	const DEPOSIT_AMOUNT: u128 = 50000;
+
+	new_test_ext().execute_with(|| {
+		let (_id, address, ..) =
+			IngressEgress::request_liquidity_deposit_address(ALICE, ASSET, 0).unwrap();
+		let deposit_address: <Ethereum as Chain>::ChainAccount = address.try_into().unwrap();
+
+		let deposit_detail: DepositWitness<Ethereum> = DepositWitness::<Ethereum> {
+			deposit_address,
+			asset: ASSET,
+			amount: DEPOSIT_AMOUNT,
+			deposit_details: (),
+		};
+
+		let deposit_witnesses = vec![deposit_detail.clone()];
+
+		// Submit the prewitnessed deposit
+		assert_ok!(IngressEgress::add_boostable_deposits(deposit_witnesses.clone(),));
+
+		// Check that the deposit is stored in the storage
+		let deposit_id = BoostableDepositIdCounter::<Test>::get();
+		let channel_id = ChannelIdCounter::<Test>::get();
+		assert_eq!(
+			BoostableDeposits::<Test>::get(channel_id, deposit_id),
```

Comment:
We can check the Prewitness event is emitted here too

Reply:
The `Prewitnessed` event is not emitted by the `add_boostable_deposits` function. It is emitted by the `prewitness_and_execute`, so we can't test for it here.

Reply:
yeah you're right 👍